src/polyswarmd/utils.py

```diff
@@ -226,24 +230,31 @@ def camel_case_to_snake_case(s):
     return re.sub(r'([a-z0-9])([A-Z])', r'\1_\2', s1).lower()
 
 
-def to_padded_hex(val):
-    if not g.chain:
-        raise ValueError('g.chain not found')
+def to_padded_hex(val: Union[str, bool, int, bytes]) -> str:
+    """Convert an argument to a hexadecimal string and zero-extend to 64 width"""
+    def encode_hex(xs: bytes) -> str:
+        return codecs.encode(xs, "hex").decode("ascii") # type: ignore
 
-    if type(val) == str:
+    if isinstance(val, str):
         if val.startswith('0x'):
-            padded_hex = g.chain.w3.toHex(hexstr=val)[2:]
+            hex_suffix = val[2:].lower()
+            # verify we're passing back a hexadecimal value
+            if not all(c in string.hexdigits for c in hex_suffix):
+                raise ValueError("Invalid hexadecimal characters detected")
         else:
-            padded_hex = g.chain.w3.toHex(text=val)[2:]
+            hex_suffix = encode_hex(val.encode('utf-8'))
+    elif isinstance(val, bool):
+        hex_suffix = str(int(val))
+    # do we need to check other types here?
+    elif isinstance(val, (bytes, bytearray)):
+        hex_suffix = encode_hex(val)
+    elif isinstance(val, int):
+        hex_suffix = hex(val)[2:]
     else:
-        padded_hex = g.chain.w3.toHex(val)[2:]
-
-    l = 64 - len(padded_hex)
-
-    for i in range(0, l):
-        padded_hex = '0' + padded_hex
+        raise TypeError("Cannot convert to padded hex value")
 
-    return padded_hex
+    # This leaves open the possibility of returning len(result) > 64 chars.
+    return hex_suffix.rjust(64, '0')
```

Comment:
how could this return a string having len() > 64?

Reply:
@mrtizmoatwork `to_padded_hex('0x' + '1' * 65)`

Reply:
@Rizato what would happen if this function returned a string that is larger than 64 chars? So, for example. how would polyswarmd handle a nonce that is 65 or more digits long?

Reply:
I think we should raise some exception if the conversion is over 64, so we can return 400 in the offers route. Unfortunately the state blob needs everything exactly 64 length. 

If the state values weren't coming from users I would say let's ignore it.

Reply:
Alternatively, we improve the schema in the offers route to prevent too long input. 

I prefer that route.

Reply:
@mrtizmoatwork @Rizato I'm raising a `ValueError` if the result would be longer than 64 chars